From 8c73efd7c83871b397a2173dff5d4fff35015826 Mon Sep 17 00:00:00 2001
From: Raymond Wong <infiwang@pm.me>
Date: Tue, 2 May 2023 16:37:37 +0800
Subject: [PATCH 089/156] riscv(misc): correct FUNCT3/FUNCT7 instruction field
 helper

Not in use anywhere though.

This is a WIP, archive only.
---
 src/lj_target_riscv.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lj_target_riscv.h b/src/lj_target_riscv.h
index f9fe2f27..cfda1684 100644
--- a/src/lj_target_riscv.h
+++ b/src/lj_target_riscv.h
@@ -174,8 +174,8 @@ static LJ_AINLINE uint32_t *exitstub_trace_addr_(uint32_t *p)
 #define RISCVF_S2(r)	(((r)&31) << 20)
 #define RISCVF_S3(r)	(((r)&31) << 27)
 #define RISCVF_FUNCT2(f)	(((f)&3) << 25)
-#define RISCVF_FUNCT3(f)	(((f)&3) << 12)
-#define RISCVF_FUNCT7(f)	(((f)&3) << 25)
+#define RISCVF_FUNCT3(f)	(((f)&7) << 12)
+#define RISCVF_FUNCT7(f)	(((f)&127) << 25)
 #define RISCVF_SHAMT(s)	((s) << 20)
 #define RISCVF_RM(m)	(((m)&7) << 12)
 #define RISCVF_IMMI(i)	((i) << 20)
-- 
2.42.0

